frontend/src/app/modules/user-profile/user-details/user-details/user-details.component.ts

```diff
@@ -111,15 +132,17 @@ export class UserDetailsComponent extends BaseComponent implements OnInit {
         return this.detailsForm.get('email') as FormControl;
     }
 
-    get instagram(): FormControl {
-        return this.detailsForm.get('instagram') as FormControl;
+    selectInterest(tag: ITag) {
+        this.selectedTags = (this.includesTags(tag.id))
+            ? this.selectedTags.filter(x => x.id !== tag.id)
+            : [...this.selectedTags, tag];
     }
 
-    get facebook(): FormControl {
-        return this.detailsForm.get('facebook') as FormControl;
+    getIconById(id: number) {
+        return this.tagsList.find(t => t.id === id)?.link;
     }
 
-    get other(): FormControl {
-        return this.detailsForm.get('other') as FormControl;
+    includesTags(id: number) {
+        return this.selectedTags.find(t => t.id === id);
```

Comment:
two almost the same functions

Reply:
I don't understand it needs refacorring. Refactoring will be more complecated than initial solutin and later this part wel be changed

Reply:
fixed